lexicons/bsky.app/declaration.json

```diff
@@ -0,0 +1,16 @@
+{
+  "lexicon": 1,
+  "id": "app.bsky.declaration",
+  "type": "record",
+  "key": "literal:self",
+  "record": {
+    "type": "object",
+    "required": ["actorType"],
+    "properties": {
+      "actorType": {
```

Comment:
It feels a bit strange that this record is so sparse but I'm not sure if it's bad. Every account of the same type will have the same CID for their declaration.

Is there any other info we'd want to include in here?

One option would be to include `did` on the declaration as well & then a follow could be _just_ on the declaration CID. The tradeoff being that it makes follow records a decent bit more opaque which is why I didn't go with that option.

Reply:
Sparseness doesnt bother me. This is kind of a special record. If we think of other things that need this kind of behavior, we can add them.

We could include `did` but how would that change follow records? You'd still need the did of the repo in there, right?

Reply:
I personally like how you have it as-is: it feels symmetrical with how we treat uri/cid pairs on likes, reposts, etc.

Reply:
if you're aware of the declared repo, you actually _don't_ need the `did` to track the follow.

but I realized after posting this comment that you _do_ need the `did` to make the subscription to the correct repo & if you don't have it, then the CID is just this random hash you can't easily resolve

Cool i'll leave as is. It is a pretty special record so I think it's fine for it to be unique in a couple of ways (sparse & same CID for many repos)